### ISSUE TITLE
[UI] It takes long time to show the pipeline config if there are many branches

### ISSUE THREAD

**Issue by gitlawr** [rancher/rancher#22049]
UI ticket for https://github.com/rancher/rancher/issues/17231

When users Edit pipeline configs in UI, or view/edit pipeline file by yaml, UI talks to the following APIs to fetch the data:
1. `/v3/project/<id>/pipelines/<id>/branches` to get all branches.
2. `/v3/project/<id>/pipelines/<id>/yaml?branch=<name>` to get pipeline file for a specific branch.
3. `/v3/project/<id>/pipelines/<id>/configs?branch=<name>` to get pipeline config for a specific branch.

**Comment by davidnuzik** [comment]
**Reproduction:**
v2.2.7

I can't quite "reproduce" the same as described in this version. We don't hit all the same endpoints. HOWEVER, in v2.2.7 it does get quite slow when I have say 20 branches (up to 7 seconds to load edit pipeline config). So I'm going to call this a win and claim I have reproduced.

**Validation:**
master-head (v2.3) (8/16/19)

I followed the steps and can see in the latest (master-head) we hit the various endpoints mentioned. I noticed, compared to my v2.2.7 reproduction test they UI loads much, much faster now when we try to Edit Pipeline Config. I never observed the issue editing YAML in v2.2.7, but this too works in a fast manner in the master-head UI.

I didn't identify any issues during my tests.
